spec/helpers/connections.rb

```diff
@@ -9,6 +9,22 @@ def local_connection
       Etcdv3.new(url: "http://#{local_url}")
     end
 
+    def local_connection_with_tls_server_auth(cacert)
+      Etcdv3.new(url: "http://#{local_url}", cacert: cacert)
+    end
+
+    def local_connection_with_tls_client_auth(cacert, key, cert)
+      Etcdv3.new(url: "http://#{local_url}", cacert: cacert, key: key, cert: cert)
```

Comment:
Yeah, I am not sure off hand how to properly test this... We could certainly generate self-signed certs to use, but I think the tricky part is spinning up the Etcd with and without the tls parameters.